examples/run/openmp/io_dec_par_example.cpp

```diff
@@ -29,64 +28,36 @@ traccc::demonstrator_result run(traccc::demonstrator_input input_data,
-    traccc::spacepoint_formation sp(resource);
-
-    auto startAlgorithms = std::chrono::system_clock::now();
+    traccc::clusterization_algorithm ca(resource);
+    traccc::spacepoint_formation sf(resource);
 
     // Output stats
-    int64_t n_modules = 0;
+    uint64_t n_modules = 0;
     uint64_t n_cells = 0;
-    uint64_t n_clusters = 0;
     uint64_t n_measurements = 0;
-    uint64_t n_space_points = 0;
+    uint64_t n_spacepoints = 0;
+
+    auto startAlgorithms = std::chrono::system_clock::now();
 
     traccc::demonstrator_result aggregated_results(input_data.size(),
                                                    &resource);
 
-#pragma omp parallel for reduction(+:n_modules, n_cells, n_clusters, n_measurements, n_space_points)
+#pragma omp parallel for reduction (+:n_modules, n_cells, n_measurements, n_spacepoints)
     for (size_t event = 0; event < input_data.size(); ++event) {
         traccc::host_cell_container cells_per_event =
             input_data.operator[](event);
 
-        traccc::host_measurement_container measurements_per_event;
-        traccc::host_spacepoint_container spacepoints_per_event;
-        measurements_per_event.reserve(cells_per_event.size());
-        spacepoints_per_event.reserve(cells_per_event.size());
-
-#pragma omp parallel for
-        for (size_t i = 0; i < cells_per_event.size(); ++i) {
-            auto &module = cells_per_event.get_headers()[i];
-            module.pixel = traccc::pixel_data{-8.425, -36.025, 0.05, 0.05};
-
-            // The algorithmic code part: start
-            traccc::host_cluster_container clusters =
-                cc(cells_per_event.get_items()[i],
-                   cells_per_event.get_headers()[i]);
-            for (auto &cl_id : clusters.get_headers()) {
-                cl_id.pixel = module.pixel;
-            }
-
-            traccc::host_measurement_collection measurements_per_module =
-                mt(clusters, module);
-            traccc::host_spacepoint_collection spacepoints_per_module =
-                sp(module, measurements_per_module);
-            // The algorithmnic code part: end
-
-            n_cells += cells_per_event.get_items()[i].size();
-            n_clusters += clusters.size();
-            n_measurements += measurements_per_module.size();
-            n_space_points += spacepoints_per_module.size();
-
-#pragma omp critical
-            {
-                measurements_per_event.push_back(
-                    module, std::move(measurements_per_module));
-                spacepoints_per_event.push_back(
-                    module.module, std::move(spacepoints_per_module));
-            }
-        }
+        /*-------------------
+            Clusterization
+          -------------------*/
+
+        auto measurements_per_event = ca(cells_per_event);
+
+        /*------------------------
+            Spacepoint formation
+          ------------------------*/
+
+        auto spacepoints_per_event = sf(measurements_per_event);
 
```

Comment:
Here is missing the update for the counters (`n_modules, n_cells, n_measurements, n_spacepoints`) and a critical region to aggregate the measurements and spacepoints.

Reply:
It's fixed now